src/StoreKeeper/WooCommerce/B2C/Commands/ScheduledProcessor.php

```diff
@@ -13,23 +15,20 @@ class ScheduledProcessor extends ProcessAllTasks
      */
     public function execute(array $arguments, array $assoc_arguments)
     {
-        CronOptions::set(CronOptions::LAST_PRE_EXECUTION_DATE, date('Y-m-d H:i:s'));
-        $beforeCount = ProcessAllTasks::countTasks();
+        $beforeCount = ProcessAllTasks::countNewTasks();
         try {
+            CronRegistrar::validateRunner(CronRegistrar::RUNNER_CRONTAB_CLI);
+            CronOptions::set(CronOptions::LAST_PRE_EXECUTION_DATE, date(DATE_RFC2822));
             parent::execute($arguments, $assoc_arguments);
-            CronOptions::set(CronOptions::LAST_EXECUTION_STATUS, CronRegistrar::STATUS_SUCCESS);
-            CronOptions::delete(CronOptions::LAST_POST_EXECUTION_ERROR);
+            CronOptions::updateSuccessfulExecution();
+        } catch (InvalidRunnerException $invalidRunnerException) {
+            $beforeCount = 0;
+            throw new WpCliException("Error: {$invalidRunnerException->getMessage()}");
         } catch (Throwable $throwable) {
-            CronOptions::set(CronOptions::LAST_EXECUTION_STATUS, CronRegistrar::STATUS_FAILED);
-            CronOptions::set(CronOptions::LAST_POST_EXECUTION_ERROR, $throwable->getMessage());
+            CronOptions::updateFailedExecution($throwable);
```

Comment:
how do you know if it processed any tasks, when it gets here?

Reply:
Now using wordpress transients, to save temporarily if all tasks are processed correctly, or there are errors on `ProcessAllTasks`